src/record-count/index.ts

```diff
@@ -11,10 +11,12 @@ class RecordCount {
     }
   };
   state: RecordCount.State = {
+    total: this.select(Selectors.recordCount),
     query: this.select(Selectors.currentQuery)
   };
 
   init() {
+    this.updatePageRange(this.select(Selectors.pageObject));
```

Comment:
only page range is affected by racing? Not record count or query?

Reply:
Done

Reply:
Isn't calling all three of these going to cause 3 updates? Should probably not do that.

Reply:
Oh actually the original code was correct, record count and query were both set in the constructor and thus we only need to update page range.

Reply:
Then why don't you do this in the same place as on line 14 and 15?
In some other components, state from flux are initiated in a constructor. I don't really know its difference from doing that and doing this.